### PR TITLE
[BK-1.14][FLINK-26645][Connector/pulsar] Support subscribe only one topic partition/

### PR DESCRIPTION
**Pull request by syhily**
This is a backport PR for #19092

### CODE REVIEW

**Comment by syhily** [comment]
@AHeise This is a backport PR. Can you review it?

File: flink-connectors/flink-connector-pulsar/src/test/resources/containers/txnStandalone.conf [hunk 1]
```diff
@@ -14,8 +14,7 @@
 
 ### --- General broker settings --- ###
 
-# Zookeeper quorum connection string
-zookeeperServers=
+# Zookeeper quorum connection stringzookeeperServers=
```
Comment (fapaul) [inline flink-connectors/flink-connector-pulsar/src/test/resources/containers/txnStandalone.conf:17]: I did not find this change in the original PR, and it looks suspicious.

File: flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/enumerator/subscriber/impl/BasePulsarSubscriber.java [hunk 2]
```diff
@@ -63,18 +63,15 @@ protected List<TopicPartition> toTopicPartitions(
...
-                            partitionId ->
-                                    ranges.stream()
-                                            .map(
-                                                    range ->
-                                                            new TopicPartition(
-                                                                    metadata.getName(),
-                                                                    partitionId,
-                                                                    range)))
-                    .collect(toList());
+            List<TopicPartition> partitions = new ArrayList<>();
```
Comment (fapaul) [inline flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/enumerator/subscriber/impl/BasePulsarSubscriber.java:66]: Nit: Why did you change using the stream API to the manual loop?

**Review by fapaul — COMMENTED** [review]
Looks mostly like the original PR, but one configuration change I did not understand.

File: flink-connectors/flink-connector-pulsar/src/test/resources/containers/txnStandalone.conf (hunk 1 above)
Comment (AHeise) [reply flink-connectors/flink-connector-pulsar/src/test/resources/containers/txnStandalone.conf:17]: This looks indeed as an unintended change.

Comment (syhily) [reply flink-connectors/flink-connector-pulsar/src/test/resources/containers/txnStandalone.conf:17]: Yep. This should be a mistake on my side.

File: flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/enumerator/subscriber/impl/BasePulsarSubscriber.java (hunk 2 above)
Comment (syhily) [reply flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/enumerator/subscriber/impl/BasePulsarSubscriber.java:66]: This would make the code more readable because the lambda has been formatted in an ugly style.

**Comment by imaffe** [comment]
@MartijnVisser  seems this backport PR is ready ?

**Review by MartijnVisser — APPROVED** [review]
LGTM
